app/models/learning_unit_succession.rb

```diff
@@ -0,0 +1,16 @@
+class LearningUnitSuccession < ApplicationRecord
+  belongs_to :predecessor, class_name: 'LearningUnit'
+  belongs_to :successor, class_name: 'LearningUnit'
```

Comment:
I think you could if you needed it, but I'd wait for something to need it in order to create the access.
Anyhow, taking into account that a predecessor or a successor is not an absolute relation for a learning unit (it is relative to a cycle), I don't think it would be a very good thing to call `learning_unit.predecessors` without the understanding of the cycle in which the relations are established.

Reply:
Makes sense. However, remember we are considering just one curriculum for now. So, if the relation is needed, you could create the relation and refactor when another curriculum is created.